### ISSUE TITLE
[Issue] Can't add SMTP account on fresh install

### ISSUE THREAD

**Issue by KAW0** [babelouest/glewlwyd#191]
Every time I try to add an SMTP account, the following message appears:"Error while adding SMPT configuration"

**System**
 - OS/Environment:, Ubuntu 20.04.3
 - Browser used: Chrome
 - Glewlwyd Version: 2.6.1
 - Source installation: Docker
 - Mariadb

![obraz](https://user-images.githubusercontent.com/16373327/148950441-47181331-10ba-48ef-8ee1-d84e5613ba18.png)
![obraz](https://user-images.githubusercontent.com/16373327/148950461-1df9f515-de8e-4fee-821d-0fb5fc6254f2.png)
![obraz](https://user-images.githubusercontent.com/16373327/148951724-d2863445-3990-4c61-b3a6-b2cf24b8078b.png)

**Comment by babelouest** [comment]
Hello @KAW0 ,

First, beware that you seem to use the [docker development version](https://github.com/babelouest/glewlwyd/blob/master/docs/INSTALL.md#docker-image-builder), not the [2.6.1 release](https://github.com/babelouest/glewlwyd/blob/master/docs/INSTALL.md#quick-start-for-tests-only), because the misc configuration page will be in the next 2.7 version.

I suppose you used the wrong sql script to install the database, that's why you have the error message, because your  input data looks good, and I couldn't reproduce your error.

You have 2 options then:
- Pull and run the [official 2.6.1 docker instance](https://hub.docker.com/r/babelouest/glewlwyd) with a database initialized with the [2.6.1 release](https://github.com/babelouest/glewlwyd/blob/v2.6.1/docs/database/init.mariadb.sql)
- Build your [glewlwyd docker image](https://github.com/babelouest/glewlwyd/blob/master/docs/INSTALL.md#docker-image-builder) from current source, using the [development mariadb init script](https://github.com/babelouest/glewlwyd/blob/master/docs/database/init.mariadb.sql)

**Comment by KAW0** [comment]
> I suppose you used the wrong sql script to install the database, that's why you have the error message, because your input data looks good, and I couldn't reproduce your error.

It's true, but I figured it out before writing this topic then I purget database and insert right sql however nothing has changed.
I just built a new docker image from newest version and it's works as expected.

About Docker image I'm afraid 2.6.1 and the newest are the same.
![obraz](https://user-images.githubusercontent.com/16373327/148967271-e41bf41c-b949-4f84-9f3a-d9bc5746e36a.png)

**Comment by KAW0** [comment]
Sorry but you don't understand.
I repeated the whole process on VM and I can confirm adding email doesn't work on 2.6.1 at all even with correct sql (from here: https://github.com/babelouest/glewlwyd/archive/refs/tags/v2.6.1.zip)

**Comment by babelouest** [comment]
Sorry for the misunderstanding, can you send the log error messages from the console and browser then?

**Comment by KAW0** [comment]
I can even provide a virtualbox image if needed (on pw), but uploading can take a while.
Browser:
> /auth/locales/pl-PL/translations.json:1 Failed to load resource: the server responded with a status of 404 (Not Found)
/auth/locales/pl/translations.json:1 Failed to load resource: the server responded with a status of 404 (Not Found)
/auth/api/misc:1 Failed to load resource: the server responded with a status of 404 (Not Found)
192.168.2.194/:1 [DOM] Found 4 elements with non-unique id #btnGroupNavPerPage: (More info: https://goo.gl/9p2vKq) <button id=​"btnGroupNavPerPage" type=​"button" class=​"btn btn-secondary dropdown-toggle" data-toggle=​"dropdown" aria-haspopup=​"true" aria-expanded=​"false">​Results per page​</button>​ <button id=​"btnGroupNavPerPage" type=​"button" class=​"btn btn-secondary dropdown-toggle" data-toggle=​"dropdown" aria-haspopup=​"true" aria-expanded=​"false">​Results per page​</button>​ <button id=​"btnGroupNavPerPage" type=​"button" class=​"btn btn-secondary dropdown-toggle" data-toggle=​"dropdown" aria-haspopup=​"true" aria-expanded=​"false">​Results per page​</button>​ <button id=​"btnGroupNavPerPage" type=​"button" class=​"btn btn-secondary dropdown-toggle" data-toggle=​"dropdown" aria-haspopup=​"true" aria-expanded=​"false">​Results per page​</button>​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-content-type-from: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-content-type-from" placeholder=​"Content-Type:​ text/​plain;​ charset=utf-8" value=​"text/​plain;​ charset=utf-8">​ <input type=​"text" class=​"form-control" id=​"mod-content-type-from" placeholder=​"Content-Type:​ text/​plain;​ charset=utf-8" value=​"text/​plain;​ charset=utf-8">​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-display-name: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-display-name" placeholder=​"e.g. Database" maxlength=​"256" value>​ <input type=​"text" class=​"form-control" id=​"mod-display-name" placeholder=​"e.g. Database" maxlength=​"256" value>​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-email-from: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-email-from" placeholder=​"e.g. noreply@glewlwyd.tld" value=​"ajugapl@gmail.com">​ <input type=​"text" class=​"form-control" id=​"mod-email-from" placeholder=​"e.g. noreply@glewlwyd.tld" value>​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-email-password: (More info: https://goo.gl/9p2vKq) <input type=​"password" class=​"form-control" id=​"mod-email-password" placeholder=​"e.g. password" value=​"3gvaY6qQBI7XbfxNjAXq">​ <input type=​"password" class=​"form-control" id=​"mod-email-password" placeholder=​"e.g. password" value>​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-email-port: (More info: https://goo.gl/9p2vKq) <input type=​"number" min=​"0" max=​"65536" step=​"1" class=​"form-control" id=​"mod-email-port" placeholder=​"e.g. 25" value=​"465">​ <input type=​"number" min=​"0" max=​"65536" step=​"1" class=​"form-control" id=​"mod-email-port" placeholder=​"e.g. 25" value=​"25">​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-email-user: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-email-user" placeholder=​"e.g. mail-user" value=​"ajugapl@gmail.com">​ <input type=​"text" class=​"form-control" id=​"mod-email-user" placeholder=​"e.g. mail-user" value>​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-email-user-lang-property: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-email-user-lang-property" placeholder=​"e.g. lang" value=​"lang">​ <input type=​"text" class=​"form-control" id=​"mod-email-user-lang-property" placeholder=​"e.g. lang" value=​"lang">​
192.168.2.194/:1 [DOM] Found 2 elements with non-unique id #mod-name: (More info: https://goo.gl/9p2vKq) <input type=​"text" class=​"form-control" id=​"mod-name" placeholder=​"e.g. db" maxlength=​"128" disabled value>​ <input type=​"text" class=​"form-control" id=​"mod-name" placeholder=​"e.g. db" maxlength=​"128" disabled value>​
/auth/api/misc/smtp-txz0p99bav:1 Failed to load resource: the server responded with a status of 404 (Not Found)
DevTools failed to load SourceMap: Could not load content for http://192.168.2.194/auth/js/popper.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

Console:
> 
> 2022-01-11T16:11:04Z - Glewlwyd INFO: Glewlwyd - Received close signal: Terminated
> 
> 2022-01-11T16:11:04Z - Glewlwyd INFO: Glewlwyd stopped
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Starting Glewlwyd SSO authentication service
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user module /usr/lib/glewlwyd/user/libusermodhttp.so - http
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user module /usr/lib/glewlwyd/user/libusermoddatabase.so - database
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user module /usr/lib/glewlwyd/user/libusermodldap.so - ldap
> 
> 2022-01-11T16:11:06Z - Glewlwyd ERROR: init_user_middleware_module_list - Error reading libraries folder /usr/lib/glewlwyd/user_middleware
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading client module /usr/lib/glewlwyd/client/libclientmodldap.so - ldap
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading client module /usr/lib/glewlwyd/client/libclientmoddatabase.so - database
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodotp.so - otp
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodwebauthn.so - webauthn
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodemail.so - email
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodoauth2.so - oauth2
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodhttp.so - http
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodcertificate.so - certificate
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading user auth scheme module /usr/lib/glewlwyd/scheme/libschememodpassword.so - retype-password
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading plugin module /usr/lib/glewlwyd/plugin/libprotocol_oidc.so - oidc
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading plugin module /usr/lib/glewlwyd/plugin/libprotocol_register.so - register
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Loading plugin module /usr/lib/glewlwyd/plugin/libprotocol_oauth2.so - oauth2-glewlwyd
> 
> 2022-01-11T16:11:06Z - Glewlwyd INFO: Glewlwyd started on port 4593, prefix: api, secure: false, bind address: no, external URL: http://192.168.2.194/auth

**Comment by babelouest** [comment]
To make sure we're talking about the same thing: do you want to run the last release 2.6.1 or do you want to run the current master branch?

**Comment by KAW0** [comment]
Currently, I will stick to the built I made from master branch. I'm not at the production stage yet.

**Comment by babelouest** [comment]
Did you initialize the database using the sql script from the master branch? https://github.com/babelouest/glewlwyd/blob/master/docs/database/init.mariadb.sql

If so, when you try to add an SMTP configuration, do you have errors in the console?

Also do you see a response error message on the browser console network tab on the `POST /api/misc` request?

**Comment by KAW0** [comment]
Ok. To be clear:

Currently, I have installed versions from the master branch on my server. It works as it should.

I installed 2.6.1 on my virtual machine to show that in this version adding email doesn't work and logs are from here.

I believe this is the error that this applies to (from Firefox, the previous one was from Chrome):
![obraz](https://user-images.githubusercontent.com/16373327/148990467-4711459f-4a55-42bf-aae9-518e27a0309b.png)

**Comment by babelouest** [comment]
It's weird, I couldn't reproduce your problem, even if I try to reproduce your configuration:

- Set value `external_url` to `https://localhost/auth/`
- Run glewlwyd from source on localhost:4593
- Run apache with mod_proxy_http to redirect `http://localhost:4593/` to `https://localhost/auth/`
- Open `https://localhost/auth/` and use glewlwyd from there

Even then, the HTTP request used to push the new smtp config works:
![image](https://user-images.githubusercontent.com/3966617/148999920-dc917de4-655b-48d5-bf49-943de0c19ebc.png)

Does the other misc config work? `misc-ip-geolocation-api` for example. (sorry, I forgot to add the english translation for the new page)

Also, does the rest of the admin application seem to work?

**Comment by KAW0** [comment]
Similar error: 
![obraz](https://user-images.githubusercontent.com/16373327/149003973-491881f4-8b28-4cdb-973c-f322213273ad.png)

**Comment by babelouest** [comment]
Do you have a similar error if you add a user or a client?

**Comment by KAW0** [comment]
Nope:
![obraz](https://user-images.githubusercontent.com/16373327/149004882-a2d0e409-f602-4748-b7ae-a8a72d68c42b.png)

**Comment by babelouest** [comment]
This is weird, it looks like you have the 2.6.1 backend with the master branch front-end. Can you post the script you used to build the program and with what sources?

**Comment by babelouest** [comment]
I don't understand how you have the master branch front-end application with the 2.6.1 backend API but it seems to be because of that.

In your DockerCompose file, you specify the glewlwyd image to use like this:

```
services:
  glewlwyd:
    image: babelouest/glewlwyd:latest
```

where it should be something like that:
```
services:
  glewlwyd:
    image: babelouest/glewlwyd:src
```

with a docker image `babelouest/glewlwyd:src` you built yourself as documented in the [docker image builder](https://github.com/babelouest/glewlwyd/blob/master/docs/INSTALL.md#docker-image-builder).
Basically something like

```shell
$ git clone https://github.com/babelouest/glewlwyd.git
$ cd glewlwyd
$ make docker
```

**Comment by KAW0** [comment]
Approx. Now I understand. I'm sorry for troubling you but to be honest installation process isn't very clear for those who use docker. Especially since when you look at the documentation, it will point out files that you shouldn't use. The first instinct will not be to download the source, but to look at the documentation. (and yes it was my mistake that I omit first paragraph).
I think that documentation and Docker description should point user to pay attention when they install sql and webapp.

**Comment by babelouest** [comment]
Can you send a pull request to help improve the documentation then?
That would be much appreciated, thanks in advance!

**Comment by babelouest** [comment]
Fixed in #192 , thanks @KAW0 !
